### ISSUE TITLE
Updating on Windows 7 breaks "activate"

### ISSUE THREAD

**Issue by williamgilpin** [conda/conda#5321]
On my Windows 7 machine I ran

`
conda update conda
`
The update proceeded without any errors. However, now whenever I run the activate script, I get the following terse error message

    >activate py3env
    ( was unexpected at this time.
    C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>@IF "1" == "1" @IF 
    "xCONDA_DEFAULT_ENV=PROMPT" (

I tried to revert to previous versions of conda, but that did not resolve the issue. Thank you very much for any suggestions.

**Comment by kalefranz** [comment]
What is the output of

    conda info -a

and

    set

?

**Comment by williamgilpin** [comment]
For the `conda info -a` command, I get

```
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>conda info -a
Current conda install:

               platform : win-64
          conda version : 4.3.18
       conda is private : False
      conda-env version : 4.3.18
    conda-build version : not installed
         python version : 3.4.5.final.0
       requests version : 2.14.2
       root environment : C:\Users\me\AppData\Local\Continuum\Miniconda3  (writable)
    default environment : C:\Users\me\AppData\Local\Continuum\Miniconda3
       envs directories : C:\Users\me\AppData\Local\Continuum\Miniconda3\envs
                          C:\Users\me\AppData\Local\conda\conda\envs
                          C:\Users\me\.conda\envs
          package cache : C:\Users\me\AppData\Local\Continuum\Miniconda3\pkgs
                          C:\Users\me\AppData\Local\conda\conda\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : C:\Users\me\.condarc
           offline mode : False
             user-agent : conda/4.3.18 requests/2.14.2 CPython/3.4.5 Windows/7 Windows/6.1.7601
          administrator : unknown
# conda environments:
#
py2                      C:\Users\me\AppData\Local\Continuum\Miniconda3\envs\py2
py2env                   C:\Users\me\AppData\Local\Continuum\Miniconda3\envs\py2env
py3env                   C:\Users\me\AppData\Local\Continuum\Miniconda3\envs\py3env
root                  *  C:\Users\me\AppData\Local\Continuum\Miniconda3

sys.version: 3.4.5 |Continuum Analytics, Inc.| (defau...
sys.prefix: C:\Users\me\AppData\Local\Continuum\Miniconda3
sys.executable: C:\Users\me\AppData\Local\Continuum\Miniconda3\python.exe
conda location: C:\Users\me\AppData\Local\Continuum\Miniconda3\lib\site-packages\conda
conda-build: None
conda-env: C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts\conda-env.exe
user site dirs:

CIO_TEST: <not set>
CONDA_DEFAULT_ENV: <not set>
CONDA_ENVS_PATH: C:\Users\me\AppData\Local\Continuum\Miniconda3\envs
PATH: C:\Users\me\AppData\Local\Continuum\Miniconda3\Library\bin;C:\Users\me\AppData\Local\Continuum\Miniconda
3\envs\py2;C:\Program Files\ImageMagick-7.0.1-Q16;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\Sys
tem32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\MIT\Kerberos\bin;C:\Program Files (x86)\Common Files\SYSTEM\MSMAPI\
1033;C:\Windows\SysWow64;C:\Program Files\VanDyke Software\Clients\;C:\Program Files (x86)\Common Files\Roxio Shared\OEM
\DLLShared\;C:\Program Files (x86)\Common Files\Roxio Shared\OEM\DLLShared\;C:\Program Files (x86)\Common Files\Roxio Sh
ared\OEM\12.0\DLLShared\;C:\Program Files (x86)\Roxio\OEM\AudioCore\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI
\;C:\Program Files (x86)\QuickTime\QTSystem\;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Users\me\AppData\Local\
Continuum\Miniconda3;C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts
PYTHONHOME: <not set>
PYTHONPATH: <not set>

WARNING: could not import _license.show_info
# try:
# $ conda install -n root _license

```

For `set` I get

```
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>set
ALLUSERSPROFILE=C:\ProgramData
ANACONDA_ENVS=C:\Users\me\AppData\Local\Continuum\Miniconda3\envs
APPDATA=C:\Users\me\AppData\Roaming
CLIENTNAME=DN51scrt.SUNet
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=SOE-20S3W52
ComSpec=C:\Windows\system32\cmd.exe
CONDA_ENVS_PATH=C:\Users\me\AppData\Local\Continuum\Miniconda3\envs
FP_NO_HOST_CHECK=NO
HOMEDRIVE=C:
HOMEPATH=\Users\me
LOCALAPPDATA=C:\Users\me\AppData\Local
LOGONSERVER=\\GODZILLA
NUMBER_OF_PROCESSORS=8
OS=Windows_NT
Path=C:\Users\me\AppData\Local\Continuum\Miniconda3\envs\py2;C:\Program Files\ImageMagick-7.0.1-Q16;C:\Windows\syst
em32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\MIT\Kerberos
\bin;C:\Program Files (x86)\Common Files\SYSTEM\MSMAPI\1033;C:\Windows\SysWow64;C:\Program Files\VanDyke Software\Client
s\;C:\Program Files (x86)\Common Files\Roxio Shared\OEM\DLLShared\;C:\Program Files (x86)\Common Files\Roxio Shared\OEM\
DLLShared\;C:\Program Files (x86)\Common Files\Roxio Shared\OEM\12.0\DLLShared\;C:\Program Files (x86)\Roxio\OEM\AudioCo
re\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI\;C:\Program Files (x86)\QuickTime\QTSystem\;C:\Windows\System32\
WindowsPowerShell\v1.0\;C:\Users\me\AppData\Local\Continuum\Miniconda3;C:\Users\me\AppData\Local\Continuum\Min
iconda3\Scripts
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=3c03
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PSModulePath=C:\Windows\system32\WindowsPowerShell\v1.0\Modules\;c:\Program Files\Microsoft Security Client\MpProvider\
PUBLIC=C:\Users\Public
SESSIONNAME=RDP-Tcp#0
SystemDrive=C:
SystemRoot=C:\Windows
TEMP=C:\Users\me\AppData\Local\Temp
TMP=C:\Users\me\AppData\Local\Temp
USERDNSDOMAIN=WIN.STANFORD.EDU
USERDOMAIN=WIN
USERNAME=me
USERPROFILE=C:\Users\me
windir=C:\Windows
windows_tracing_flags=3
windows_tracing_logfile=C:\BVTBin\Tests\installpackage\csilogfile.log

C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>
```

Although PYTHONPATH is not set, I am still able to open python outside of a virtual environment. I just am unable to open my existing virtual environments. Thank you very much for your help with this.

**Comment by kalefranz** [comment]
Ok thanks for that.  Now some more info...  Can you also give me the output of

    conda config --show

and

    conda config --show-sources

?

**Comment by kalefranz** [comment]
The code that appears to be failing here is in activate.bat

```
:: if our prompt var does not contain reference to CONDA_DEFAULT_ENV, set prompt
@IF "%CHANGE_PROMPT%" == "1" @IF "x%PROMPT:CONDA_DEFAULT_ENV=%" == "x%PROMPT%" (
    SET "PROMPT=(%CONDA_NEW_ENV%) %PROMPT%"
)
```

**Comment by kalefranz** [comment]
Also, after you give me the output of the conda config commands, try

    conda config --set changeps1 false

And then try activating your environment.

**Comment by williamgilpin** [comment]
Running `conda confige --show` yeilds

```

C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>conda config --show
add_anaconda_token: True
add_pip_as_python_dependency: True
allow_non_channel_urls: True
allow_softlinks: True
always_copy: False
always_softlink: False
always_yes: False
anaconda_upload: None
auto_update_conda: True
changeps1: True
channel_alias: https://conda.anaconda.org
channel_priority: True
channels:
  - defaults
client_ssl_cert: None
client_ssl_cert_key: None
clobber: False
create_default_packages: []
custom_channels:
  pkgs/pro: https://repo.continuum.io/
  pkgs/r: https://repo.continuum.io/
  pkgs/free: https://repo.continuum.io/
  pkgs/msys2: https://repo.continuum.io/
custom_multichannels:
  defaults: ["https://repo.continuum.io/pkgs/free", "https://repo.continuum.io/pkgs/r", "https://repo.continuum.io/pkgs/
pro", "https://repo.continuum.io/pkgs/msys2"]
  local: []
default_channels:
  - https://repo.continuum.io/pkgs/free
  - https://repo.continuum.io/pkgs/r
  - https://repo.continuum.io/pkgs/pro
  - https://repo.continuum.io/pkgs/msys2
disallow: []
envs_dirs:
  - C:\Users\me\AppData\Local\Continuum\Miniconda3\envs
  - C:\Users\me\AppData\Local\conda\conda\envs
  - C:\Users\me\.conda\envs
force: False
json: False
local_repodata_ttl: 1
migrated_channel_aliases: []
offline: False
path_conflict: clobber
pinned_packages: []
pkgs_dirs:
  - C:\Users\me\AppData\Local\Continuum\Miniconda3\pkgs
  - C:\Users\me\AppData\Local\conda\conda\pkgs
proxy_servers: {}
quiet: False
remote_connect_timeout_secs: 9.15
remote_max_retries: 3
remote_read_timeout_secs: 60.0
rollback_enabled: True
shortcuts: True
show_channel_urls: None
ssl_verify: False
track_features: []
use_pip: True
verbosity: 0

```

Running `conda config --show-sources` yields

```
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>conda config --show-sources
==> C:\Users\me\.condarc <==
ssl_verify: False

==> envvars <==
envs_path:
  - C:\Users\me\AppData\Local\Continuum\Miniconda3\envs

```
After seeing this output, I tried modifying my`.condarc` file to state `ssl_verify: True`, but that did not resolve the issue. I also tried removing this line, as well as deleting the file, and neither resolved the issue.

I then ran `conda config --set changeps1 false`, which resulted in a slight change to the error message:

```
>activate py3env
( was unexpected at this time.
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts>@IF "0" == "1" @IF 
"xCONDA_DEFAULT_ENV=PROMPT" (
```

I checked the `.condarc` file, and it now contains a new line stating `changeps1: false`
Thank you very much for your help with this.

**Comment by kalefranz** [comment]
It's possible the `CONDA_ENVS_PATH` environment variable is messing with some of the logic somewhere...  Try

    set CONDA_ENVS_PATH=

And see if that changes the activate behavior...

**Comment by kalefranz** [comment]
Actually, maybe you have an outdated activate.bat file.  What do you get for

    where activate

**Comment by kalefranz** [comment]
The current activate.bat has this line

    @IF "%CHANGE_PROMPT%" == "1" @IF "x%PROMPT:CONDA_DEFAULT_ENV=%" == "x%PROMPT%" (

The line you're showing in the error message is more like

    @IF "%CHANGE_PROMPT%" == "1" @IF  "xCONDA_DEFAULT_ENV=PROMPT" (

**Comment by kalefranz** [comment]
Yeah I'm confused.  After you get `where activate`, can you give here the lines that are actually in the file around line 51?  For comparison again...

```
:: if our prompt var does not contain reference to CONDA_DEFAULT_ENV, set prompt
@IF "%CHANGE_PROMPT%" == "1" @IF "x%PROMPT:CONDA_DEFAULT_ENV=%" == "x%PROMPT%" (
    SET "PROMPT=(%CONDA_NEW_ENV%) %PROMPT%"
)
```

**Comment by williamgilpin** [comment]
Unfortunately, `set CONDA_ENVS_PATH=` did not seem to resolve the issue. I tried it with `changeps1: false` and `changeps1: true` in my `.condarc` file, but I got the same error message in both cases.

Running `where activate` gave the following

```
C:\Users\me>where activate
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts\activate
C:\Users\me\AppData\Local\Continuum\Miniconda3\Scripts\activate.bat

```

To my knowledge, I have not altered `activate.bat`, and so you think this is likely from a previous release? I took a quick look at `C:\Users\wgilpin\AppData\Local\Continuum\Miniconda3\Scripts\activate.bat` and it has a "last modified" date of May 9, 2017. I took a quick look in a text editor, and it does say that

```
:: if our prompt var does not contain reference to CONDA_DEFAULT_ENV, set prompt
@IF "%CHANGE_PROMPT%" == "1" @IF "x%PROMPT:CONDA_DEFAULT_ENV=%" == "x%PROMPT%" (
    SET "PROMPT=(%CONDA_NEW_ENV%) %PROMPT%"
)
```

**Comment by kalefranz** [comment]
Well I'm confused.  I'm out of ideas at the moment.  I'm definitely not a batch script expert either though.  Hopefully someone else swings by here soon and has some other ideas...

**Comment by williamgilpin** [comment]
@kalefranz Thank you very much for your help. I will try reinstalling Miniconda, perhaps that will resolve the issue for now.

**Comment by mbargull** [comment]
The problem here is that the variable `PROMPT` is not set (see the `set` output). Because of this the string substitution `%VAR:str1=str2%` fails and does some funny business -- I have no idea what the heck `cmd` is doing in that case 🤣 .
Essentially one should always check whether the variable is set (e.g., `IF DEFINED VAR`) before using such substitutions.

**Comment by mbargull** [comment]
@williamgilpin I'm really puzzled how/why your `PROMPT` variable is not set. You might want to investigate the reason for that...

**Comment by williamgilpin** [comment]
@mbargull thank you very much for your help.

The rough sequence of events here was that `activate.bat` was working, then I deactivated my environment, ran `conda update conda`, and then this issue appeared---is it possible that in some circumstances updating could lead to `PROMPT` becoming unset?

The reason I ask is because I don't really use this machine for anything except the work I'm doing in conda environments, and so Windows Update would be the only other culprit that I would suspect.

In any case, I ended up deleting Miniconda and performing a fresh install,  and everything works now. Thank you very much for your help.
